packages/ag-charts-community/src/chart/series/cartesian/barSeries.ts

```diff
@@ -704,43 +707,136 @@ export class BarSeries extends AbstractBarSeries<
+            lineDash,
+            lineDashOffset,
+            seriesId,
+            stackGroup,
+            stroke,
+            strokeOpacity,
+            strokeWidth,
+            xKey,
+            yKey,
+        } satisfies CallbackParamRules<AgBarSeriesStylerParams<unknown, unknown>>;
+    }
+
+    private makeItemStylerParams(
+        dataModel: NonNullable<typeof this.dataModel>,
+        processedData: NonNullable<typeof this.processedData>,
+        datumIndex: number,
+        xValue: string,
+        isHighlight: boolean,
+        style: Required<AgBarSeriesStyle>
+    ): AgBarSeriesItemStylerParams<unknown, unknown> {
+        const { id: seriesId } = this;
+        const { xKey, yKey, stackGroup } = this.properties;
+
+        const datum = processedData.dataSources.get(seriesId)?.[datumIndex];
+        const yValue = dataModel.resolveColumnById(this, `yValue-raw`, processedData)[datumIndex];
+        const { xDomain, yDomain } = this.cachedDatumCallback('domain', () => ({
+            xDomain: this.getSeriesDomain(ChartAxisDirection.X),
+            yDomain: this.getSeriesDomain(ChartAxisDirection.Y),
+        }))!;
+        const activeHighlight = this.ctx.highlightManager?.getActiveHighlight();
+        const highlightStateString = this.getHighlightStateString(activeHighlight, isHighlight, datumIndex);
+
+        return {
+            seriesId,
+            ...datumStylerProperties(xValue, yValue, xKey, yKey, xDomain, yDomain),
+            datum,
+            xValue,
+            yValue,
+            stackGroup,
+            highlighted: isHighlight,
+            highlightState: highlightStateString,
+            ...style,
+        } satisfies CallbackParamRules<AgBarSeriesItemStylerParams<unknown, unknown>>;
+    }
+
+    private getStyle(
+        highlighted: boolean,
+        highlightState?: HighlightState
+    ): Required<AgBarSeriesStyle> & { opacity: number } {
+        const {
+            cornerRadius,
+            fill,
+            fillOpacity,
+            lineDash,
+            lineDashOffset,
+            stroke,
+            strokeOpacity,
+            strokeWidth,
+            styler,
+        } = this.properties;
+        let stylerResult: AgBarSeriesStyle = {};
+        if (styler) {
+            const stylerParams = this.makeStylerParams(highlighted, highlightState);
+            stylerResult = this.callWithContext(styler, stylerParams) ?? {};
+        }
+        return {
+            cornerRadius: stylerResult.cornerRadius ?? cornerRadius,
+            fill: stylerResult.fill ?? fill,
+            fillOpacity: stylerResult.fillOpacity ?? fillOpacity,
+            lineDash: stylerResult.lineDash ?? lineDash,
+            lineDashOffset: stylerResult.lineDashOffset ?? lineDashOffset,
+            opacity: 1,
+            stroke: stylerResult.stroke ?? stroke,
+            strokeOpacity: stylerResult.strokeOpacity ?? strokeOpacity,
+            strokeWidth: stylerResult.strokeWidth ?? strokeWidth,
```

Comment:
We should be able to avoid needing to encode the fallback `??` here, the theme processing can wrap the callback function and perform fallback handling to these values (we're using this approach in a bunch of cases to try and reduce complexity in the series) - maybe sync with @lsjroberts about how to do that?